lib/src/maintenance.dart

```diff
@@ -196,35 +196,81 @@ Future<Maintenance> detectMaintenance(
   // TODO: make these in separate steps
 
   if (!changelogExists) {
-    suggestions.add(new Suggestion.warning('Maintain `CHANGELOG.md`.',
-        'Changelog entries help clients to follow the progress in your code.'));
+    maintenanceSuggestions.add(new Suggestion.warning(
+        'Maintain `CHANGELOG.md`.',
+        'Changelog entries help clients to follow the progress in your code.',
+        penalty: new Penalty(fraction: 2000)));
   }
   if (!readmeExists) {
-    suggestions.add(new Suggestion.warning('Maintain `README.md`.',
-        'Readme should inform others about your project, what it does, and how they can use it.'));
+    maintenanceSuggestions.add(new Suggestion.warning('Maintain `README.md`.',
+        'Readme should inform others about your project, what it does, and how they can use it.',
+        penalty: new Penalty(fraction: 500)));
   }
   if (oldAnalysisOptions) {
-    suggestions.add(new Suggestion.hint('Use `analysis_options.yaml`.',
+    maintenanceSuggestions.add(new Suggestion.hint(
+        'Use `analysis_options.yaml`.',
         'Rename old `.analysis_options` file to `analysis_options.yaml`.'));
   }
   if (analysisOptionsExists && !strongModeEnabled) {
-    suggestions.add(new Suggestion.hint(
+    maintenanceSuggestions.add(new Suggestion.hint(
         'Enable strong mode analysis.',
         'Strong mode helps you to detect bugs and potential issues earlier.'
         'Start your `analysis_options.yaml` file with the following:\n\n'
         '```\nanalyzer:\n  strong-mode: true\n```\n'));
   }
 
+  final isExperimentalVersion = version.major == 0;
+  final isPreReleaseVersion = version.isPreRelease;
+
+  // Pre-v1
+  if (isExperimentalVersion) {
+    maintenanceSuggestions.add(new Suggestion.hint(
+        'Package is pre-v1 release.',
+        'While there is nothing inherently wrong with versions of `0.*.*`, it '
+        'usually means that the author is still experimenting with the generic '
+        'direction API.',
+        penalty: new Penalty(amount: 10)));
+  }
+
+  // Not a "gold" release
+  if (isPreReleaseVersion) {
+    maintenanceSuggestions.add(new Suggestion.hint(
+        'Package is pre-release.',
+        'Pre-release versions should be used with caution, their API may change '
+        'in breaking ways.',
+        penalty: new Penalty(fraction: 200)));
+  }
+
+  final errorCount = suggestions.where((s) => s.isError).length;
+  final warningCount = suggestions.where((s) => s.isWarning).length;
+  final hintCount = suggestions.where((s) => s.isHint).length;
+
+  if (errorCount > 0 || warningCount > 0) {
+    maintenanceSuggestions.add(new Suggestion.warning(
+        'Fix issues reported by `dartanalyzer`.',
+        '`dartanalyzer` reported $errorCount errors and $warningCount warnings.',
```

Comment:
The case of 1 error/warning leaves this as bad english. Use `error(s)` maybe? Or

`errors: $errorCount, warnings: $warningCount`